### PR TITLE
*: add etcd restore operator types and its related constants and auto gen files

### PR DESCRIPTION
**Pull request by fanminshi**
add etcd restore types.
add auto generated files.
add  etcd restore related constants.

all above will be used to build etcd restore operator.

### CODE REVIEW

**Comment by fanminshi** [comment]
cc/ @hongchaodeng @hasbro17

**Comment by hongchaodeng** [comment]
lgtm if test passes

File: pkg/apis/etcd/v1beta2/etcd_restore_types.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EtcdRestoreList is a list of EtcdRestore.
+type EtcdRestoreList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []EtcdBackup `json:"items"`
```
Comment (hasbro17) [inline pkg/apis/etcd/v1beta2/etcd_restore_types.go]: Isn't this supposed to be:
`EtcdRestore`

Comment (hongchaodeng) [reply pkg/apis/etcd/v1beta2/etcd_restore_types.go]: Good catch... My fault not reviewing good enough.

@fanminshi Please fix this in a follow-up PR.

**Pull request merged**